src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -353,49 +368,10 @@ private UUID getStudyEntityKeyId( UUID organizationId, UUID studyId ) {
         }
     }
 
-    private UUID getParticipantEntityKeyId( UUID organizationId, String participantId, UUID studyId ) {
-        try {
-            // load api
-            ApiClient apiClient = prodApiClientCache.get( ApiClient.class );
-            SearchApi searchApi = apiClient.getSearchApi();
-
-            // get entity set ids
-            UUID studyESID = ensureEntitySetExists( organizationId, CHRONICLE, STUDIES );
-            UUID participantsESID = ensureEntitySetExists( organizationId, CHRONICLE, PARTICIPANTS );
-            UUID participatedInESID = ensureEntitySetExists( organizationId, CHRONICLE, PARTICIPATED_IN );
-
-            // ensure study is valid
-            UUID studyEKID = Preconditions
-                    .checkNotNull( getStudyEntityKeyId( organizationId, studyId ), "study must exist" );
-
-            Map<UUID, List<NeighborEntityDetails>> neighbors = searchApi.executeFilteredEntityNeighborSearch(
-                    studyESID,
-                    new EntityNeighborsFilter(
-                            ImmutableSet.of( studyEKID ),
-                            Optional.of( Set.of( participantsESID ) ),
-                            Optional.of( ImmutableSet.of() ),
-                            Optional.of( ImmutableSet.of( participatedInESID ) )
-                    )
-            );
-
-            return neighbors
-                    .values()
-                    .stream()
-                    .flatMap( Collection::stream )
-                    .filter( neighbor -> participantId
-                            .equals( getFirstValueOrNull( neighbor.getNeighborDetails().orElse( Map.of() ),
-                                    PERSON_ID_FQN ) ) )
-                    .map( neighbor -> getFirstUUIDOrNull( neighbor.getNeighborDetails().orElse( Map.of() ), ID_FQN ) )
-                    .findFirst().orElse( null );
+    private UUID getParticipantEntityKeyId( UUID organizationId, String participantId ) {
+        Map<String, UUID> participantsById = studyParticipants.getOrDefault( organizationId, Map.of() );
 
-        } catch ( Exception e ) {
-            logger.error( "failed to get participant EKID: participantId == {}, study = {}, organizationId = {}",
-                    participantId,
-                    studyId,
-                    organizationId,
-                    e );
-            return null;
-        }
+        return participantsById.getOrDefault( participantId, null );
```

Comment:
`.get()` returns `null` by default, so I don't think you need to do `getOrDefault`.

Reply:
`.get()` will throw a NPE if the key doesn't exist unless the map permits null values, which is not the case here